worker/globalclockupdater/manifold.go

```diff
@@ -11,26 +11,32 @@ import (
 	"gopkg.in/juju/worker.v1"
 	"gopkg.in/juju/worker.v1/dependency"
 
+	"github.com/juju/juju/core/globalclock"
 	workerstate "github.com/juju/juju/worker/state"
 )
 
 // ManifoldConfig holds the information necessary to run a GlobalClockUpdater
 // worker in a dependency.Engine.
 type ManifoldConfig struct {
-	ClockName string
-	StateName string
+	ClockName        string
+	StateName        string
+	LeaseManagerName string
 
 	NewWorker      func(Config) (worker.Worker, error)
 	UpdateInterval time.Duration
 	BackoffDelay   time.Duration
+	Logger         Logger
 }
 
 func (config ManifoldConfig) Validate() error {
 	if config.ClockName == "" {
 		return errors.NotValidf("empty ClockName")
 	}
-	if config.StateName == "" {
-		return errors.NotValidf("empty StateName")
+	if config.StateName == "" && config.LeaseManagerName == "" {
```

Comment:
So why not require both and have the one worker update both at the same time?

Reply:
Covered above, I think.